reduce mangling of involved object names

q3aiml opened the pull request:
The short_name logic seems to be targeted at pods, replicasets, and
statefulsets where parts of the name can be auto generated. In other
cases the short_name mangling is undesirable as it throws away relevant
parts of the name when the name contains dashes.  For example a Service
called "my-lb" would appear in the culprit only as "my".

This is an improvement but still imperfect because 1) it is quite
possible to create many of these objects directly without any generated
suffix and 2) even when a suffix is appended it is not always 2 segments
as currently assumed. A more complete fix likely requires querying
kubernetes for details about the involved object and determining the
closest object that is not automatically created by kubernetes to
attribute the culprit to.

Commit:
reduce mangling of involved object names

The short_name logic seems to be targeted at pods, replicasets, and
statefulsets where parts of the name can be auto generated. In other
cases the short_name mangling is undesirable as it throws away relevant
parts of the name when the name contains dashes.  For example a Service
called "my-lb" would appear in the culprit only as "my".

This is an improvement but still imperfect because 1) it is quite
possible to create many of these objects directly without any generated
suffix and 2) even when a suffix is appended it is not always 2 segments
as currently assumed. A more complete fix likely requires querying
kubernetes for details about the involved object and determining the
closest object that is not automatically created by kubernetes to
attribute the culprit to.

bretthoerner commented:
Thanks for the PR.

My only fear here is that I don't currently know for sure whether `"Pod", "ReplicaSet", "StatefulSet"` is the correct set. Do you think it might be enough to make this an opt-in argument for now? So you could run yours with `MANGLE_NAMES="pod,replicaset,statefulset"`?

Commit:
make configurable which object types to mangle

Default to current behavior and make change opt-in in case the Pod,
ReplicaSet, and StatefulSet trio is not correct for everyone.

q3aiml commented:
Thanks for looking. That seems reasonable. How does this change look?

bretthoerner commented on sentry-kubernetes.py:
```diff
@@ -116,16 +117,19 @@ def watch_loop():
-            bits = name.split('-')
-            if len(bits) in (1, 2):
-                short_name = bits[0]
+            if kind in MANGLE_NAMES:
```
I think we'd want this to be `if not MANGLE_NAMES or kind in MANGLE_NAMES:` (in other words: default to the existing style) because otherwise all existing users will have to set the names they want to mangle. This would make it an opt-in thing, does that make sense or am I reading it wrong?

q3aiml replied:
Ahh, you are exactly right. Sorry about that.

Commit:
make opt-in actually opt-in

bretthoerner commented:
Thanks, I pushed up a new container.
